### PR TITLE
[DOC] Update to the using the topic operator section

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

**Documentation**
Recreated the _Using the Topic Operator_ chapter, so it's consistent with the recent update to the _Using the User Operator_ section.

We now have a single _configuring_ procedure.

Removed 3 files

**Old structure**

5. Using the Topic Operator
5.1. Topic Operator usage recommendations
5.2. Creating a topic
5.3. Changing a topic
5.4. Deleting a topic

**New structure**

5. Using the Topic Operator
5.1. Kafka topic resource
5.1.1. Kafka topic usage recommendations
5.1.2. Kafka topic naming conventions
5.2. Configuring a Kafka topic

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: documentation/modules/proc-configuring-kafka-topic.adoc [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+** Changing topic names using `spec.topicName`
+** Decreasing partition size using `spec.partitions`
+* You cannot change the number of replicas initially specified using `spec.replicas`.
+* Increasing `spec.partitions` for topics with keys will change how records are partitioned, which can be particularly problematic when the topic uses _semantic partitioning_.
+
+.Prerequisites
+
+* A running Kafka cluster xref:con-mutual-tls-authentication-deployment-configuration-kafka[configured with a Kafka broker listener using TLS authentication and encryption].
+* A running Topic Operator (typically xref:assembly-kafka-entity-operator-deployment-configuration-kafka[deployed with the Entity Operator]).
+* For deleting a topic, `delete.topic.enable=true` (default)
```
Comment (ppatierno) [inline documentation/modules/proc-configuring-kafka-topic.adoc:33]: Should we clarify that this configuration is in the `spec.kafka.config` in the `Kafka` resource?

File: documentation/modules/ref-topic-operator.adoc [hunk 2]
```diff
@@ -2,23 +2,32 @@
...
 NOTE: Recommendations for identifiers and names in Kubernetes are outlined in link:https://github.com/kubernetes/community/blob/master/contributors/design-proposals/architecture/identifiers.md[Identifiers and Names in Kubernetes^] community article.
 
-.Kafka topic naming conventions
+== Kafka topic naming conventions
 
 Kafka and Kubernetes impose their own validation rules for the naming of topics in Kafka and `KafkaTopic.metadata.name` respectively.
 There are valid names for each which are invalid in the other.
 
-Using the `spec.topicName property`, it is possible to create a valid topic in Kafka with a name that would be invalid for the KafkaTopic in Kubernetes.
+Using the `spec.topicName property`, it is possible to create a valid topic in Kafka with a name that would be invalid for the Kafka topic in Kubernetes.
```
Comment (ppatierno) [inline documentation/modules/ref-topic-operator.adoc:30]: `spec.topicName` property

File: documentation/modules/proc-configuring-kafka-topic.adoc (hunk 1 above)
Comment (PaulRMellor) [reply documentation/modules/proc-configuring-kafka-topic.adoc:33]: For deleting a topic, `delete.topic.enable=true` (default) in the `spec.kafka.config` of the `Kafka` resource.

File: documentation/modules/proc-configuring-kafka-topic.adoc [hunk 3]
```diff
@@ -0,0 +1,58 @@
+// Module included in the following assemblies:
+//
+// assembly-using-the-topic-operator.adoc
+
+[id='proc-configuring-kafka-topic-{context}']
+= Configuring a Kafka topic
+
+Use the properties of the `KafkaTopic` resource to configure a Kafka topic.
+
+You can use `kubectl apply` to create or modify topics, and kubectl delete to delete existing topics.
```
Comment (laidan6000) [inline documentation/modules/proc-configuring-kafka-topic.adoc:10]: Missing literals for second command.

File: documentation/modules/proc-configuring-kafka-topic.adoc [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+This procedure shows how to create a topic with 10 partitions and 2 replicas.
+
+.Before you start
+
+It is important that you consider the following before making your changes:
+
+* Kafka does _not_ support making the following changes through the `KafkaTopic` resource:
+** Changing topic names using `spec.topicName`
+** Decreasing partition size using `spec.partitions`
+* You cannot change the number of replicas initially specified using `spec.replicas`.
```
Comment (laidan6000) [inline documentation/modules/proc-configuring-kafka-topic.adoc:26]:
```suggestion
* You cannot change the number of replicas that were initially specified using `spec.replicas`.
```

**Pull request merged**